proposals/NNNN-byte-address-buffer-alignment.md

```diff
@@ -0,0 +1,983 @@
+* **Implementation Dependent**: The actual behavior depends on hardware and driver implementation details
+* **Potential Consequences**: May include incorrect results, performance degradation, or in extreme cases, hardware
+  exceptions
+
+### Runtime Additions
+
+#### Runtime Information
+
+This proposal leverages existing DXIL infrastructure and requires minimal additional runtime processing beyond what is
+already provided by the Direct3D 12 runtime and driver stack.
+
+##### Compiler Requirements
+
+The compiler must provide the following information to the runtime for proper buffer alignment support:
+
+* **DXIL Buffer Operation Alignment**: The compiler populates existing alignment parameters in DXIL buffer operations
+  (`dx.op.rawBuffer[Vector]Load.*`, `dx.op.rawBuffer[Vector]Store.*`) with **absolute alignment** values calculated from
+  multiple scenarios:
+  * **AlignedLoad/AlignedStore with BaseAlignment**: DXC must convert the relative offset alignment from function
+    parameters to absolute alignment by computing `min(BaseAlignment, function_alignment_parameter)`
+  * **Existing Load/Store with BaseAlignment**: DXC calculates absolute alignment using the "largest scalar type
+    contained in the given aggregate type" as the implied alignment argument, combined with the buffer's `BaseAlignment`
+    via `min(BaseAlignment, largest_scalar_type_size)`
+  * **AlignedLoad/AlignedStore without BaseAlignment**: DXC uses existing HLSL specification alignment requirements
+    based on the largest scalar type size, with error checking for alignment parameter mismatches
+  * **Format**: Standard DXIL buffer operation alignment parameters (expecting absolute alignment)
+  * **Runtime Usage**: Backend compilers can use these parameters for vectorization and memory access optimization
+
+* **DXIL Resource Properties Alignment**: The compiler populates the existing `BaseAlignLog2` field in
+  `dx.types.ResourceProperties` during `dx.op.annotateHandle` operations with the declared `BaseAlignment` attribute
+  values
+  * **Format**: Standard DXIL resource metadata structures
+  * **Runtime Usage**: Runtime and backend compilers can use this information for resource binding validation and
+    optimization
+
+##### Optional Shader Flag Addition
+
+The runtime may benefit from a new shader flag to optimize handling of shaders that use buffer alignment features:
+
+```cpp
+// Proposed addition to D3D_SHADER_FLAGS
+#define D3D_SHADER_FLAG_USES_BUFFER_OBJECT_ALIGNMENT 0x00200000
+```
+
+**Benefits:**
+
+* **Performance Optimization**: Runtimes can enable alignment-specific optimization paths only for shaders that use the
+  feature
+* **Resource Efficiency**: Allows runtimes to skip alignment-related processing overhead for shaders that don't use
+  alignment features
+* **Validation Efficiency**: GPU-Based Validation can selectively enable alignment checking only when necessary
+
+##### Runtime Processing Requirements
```

Comment:
From this section through `Performance Characteristics`, there's a lot of redundancy with previous content, and these sections don't fit into the rest of the document structure nicely. Do these sections even contain any new information necessary for this proposal?